tests/Tests/500__jupyterhub/test-jupyterlab-git.robot

```diff
@@ -0,0 +1,162 @@
+*** Settings ***
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Library          OpenShiftCLI
+Library          DebugLibrary
+Suite Setup      Server Setup
+Suite Teardown   End Web Test
+
+
+*** Variables ***
+${REPO_URL} =    *****
+${DIR_NAME} =  Python
+${FILE_PATH} =   Python/file.ipynb
+${EMAIL_ID} =  user@gmail.com
+${NAME} =  user
+${COMMIT_MSG} =  commit msg2
+
+
+*** Test Cases ***
+
+Verify Pushing Project Changes Remote Repository
+    [Tags]  ODS-326
+    ...     Tier1
+    ${randnum}=  Generate Random String  9  [NUMBERS]
+    ${commit_message} =    Catenate    ${COMMIT_MSG}    ${randnum}
+    Push Some Changes to Repo    ${GITHUB_USER.USERNAME}  ${GITHUB_USER.TOKEN}   ${FILE_PATH}    ${REPO_URL}    ${commit_message}
+
+Verify updating your project with changes from a git repository
+  [Tags]    ODS-324
+  ...       Tier1
+  Clone Git Repository And Open    ${REPO_URL}    ${FILE_PATH}
+  Sleep    1s
+  Open With JupyterLab Menu    File    New    Notebook
+  Sleep    2s
+  Maybe Close Popup
+  Close Other JupyterLab Tabs
+  Maybe Close Popup
+  Sleep    1
+
+  ${commit_msg1}=  Run Cell And Get Output    !git log --name-status HEAD^..HEAD | sed -n 5p
+  
+
+  Add and Run JupyterLab Code Cell in Active Notebook  ! mkdir ../folder/
+  
+
+  Sleep    5s
+
+  Open Folder or File    folder
+  
+  ${randnum}=  Generate Random String  9  [NUMBERS]
+  ${commit_message} =    Catenate    ${COMMIT_MSG}    ${randnum}
+  #now do here some changes
+  Push Some Changes to Repo    ${GITHUB_USER.USERNAME}    ${GITHUB_USER.TOKEN}    folder/${FILE_PATH}    ${REPO_URL}    ${commit_message}
+  #go to previous dir
+  Close All JupyterLab Tabs
+
+  Open Folder or File    ${DIR_NAME}
+
+  Open With JupyterLab Menu    Git    Pull from Remote
+  Sleep    2s
+  Open With JupyterLab Menu    File    New    Notebook
+  Sleep    2s
+  Maybe Close Popup
+  Close Other JupyterLab Tabs
+  Maybe Close Popup
+  Sleep    1
+  ${commit_msg2}=  Run Cell And Get Output    !git log --name-status HEAD^..HEAD | sed -n 5p
+  Should Not Be Equal    ${commit_msg2}    ${commit_msg1}
+
+
+
+*** Keywords ***
+Server Setup
+    [Documentation]    Suite Setup
+    Begin Web Test
+    Launch JupyterHub Spawner From Dashboard
+    Spawn Notebook With Arguments    image=s2i-minimal-notebook    size=Default
+
+
+Push Some Changes to Repo
+    [Arguments]  ${github username}  ${token}   ${filepath}    ${githublink}    ${commitmsgg}
+
+    Clone Git Repository In Current Folder      ${githublink}
+
+    Open Folder or File    ${filepath}
+
+    Maybe Close Popup
+    Sleep    2s
+    Wait Until JupyterLab Code Cell Is Not Active
+    Run Cell And Get Output  print("Hi \\n Hello")
+    Sleep    1s
+    Add and Run JupyterLab Code Cell in Active Notebook  print("Hi Hello")
+    Sleep    2s
```

Comment:
I think one print is enough

Reply:
I tried to use one print but it was not working, that's why I used two print stmts